### PR TITLE
Update MacOSX install instructions.

### PR DESCRIPTION
**Pull request by spco**
- Includes instructions to ensure Xcode and command line tools are up-to-date.
- Adds lines to ensure the correct compilers are picked up.
- Includes instructions to generate and run an Aspect project in Xcode in both serial and parallel.

I'd appreciate your thoughts on what I've written.

There is also the matter of the link to Matsui's instructions (URL on line 2025) which are now out-of-date - should I remove the link altogether, or still provide it but caution that it is out-of-date?

### CODE REVIEW

File: doc/manual/manual.tex [hunk 1]
```diff
@@ -2022,9 +2022,17 @@ \subsection{Compiling a static \aspect{} executable}
...
 
 \subsection{Installing and running \aspect{} on Mac OS X}
-OS X has some eccentricities which can complicate installation of \aspect{}. Currently the easiest and most reliable way to run \aspect{} under Mac OS X Mavericks (10.9) and Yosemite (10.10) is to install and run the binary package for \dealii{}. The step-by-step process is described in detail, with screenshots, here: \url{https://wiki.geodynamics.org/_media/software:aspect:aspect_yosemite_20150529.pdf}
+OS X has some eccentricities which can complicate installation of \aspect{}. Currently the easiest and most reliable way to run \aspect{} under Mac OS X Mavericks (10.9), Yosemite (10.10) and El Capitan (10.11) is to install and run the binary package for \dealii{}. The step-by-step process is described in detail, with screenshots, here: \url{https://wiki.geodynamics.org/_media/software:aspect:aspect_yosemite_20150529.pdf}
 
 \begin{enumerate}
+\item Install xcode from the app store. You might need to install the command line tools. Open a terminal and make sure ``clang'' does not report ``command not found''. You might need to run
```
Comment (bangerth) [inline doc/manual/manual.tex]: I think the correct spelling is Xcode, right?

Comment (bangerth) [inline doc/manual/manual.tex]: Do we have guidance which exact versions of the compilers so installed we need? We've had perpetual problems if the locally installed compilers are different from the ones used to compile the .dmg.

File: doc/manual/manual.tex [hunk 2]
```diff
@@ -2045,18 +2053,26 @@ \subsection{Installing and running \aspect{} on Mac OS X}
 \end{verbatim}}
 \dealii{} app opens a terminal window and displays a \dealii{} message.   \dealii{}.app will install all required libraries for \aspect{} (p4est, parMeTiS, and Trilinos) and will include the environment variables needed to use these libraries.
 
+To ensure the correct compilers are picked up when configuring \aspect{}, add the following lines to your .profile or .bash\_profile:
```
Comment (bangerth) [inline doc/manual/manual.tex]: mark up these file names with `\texttt{...}`. Maybe qualify these names with `~/` to indicate that they're in the home dir?

File: doc/manual/manual.tex [hunk 3]
```diff
@@ -2045,18 +2053,26 @@ \subsection{Installing and running \aspect{} on Mac OS X}
...
+
+\begin{verbatim}
+export OMPI_CXX=clang++
+export OMPI_CC=clang
```
Comment (bangerth) [inline doc/manual/manual.tex]: for consistency with other places, indent these commands by two spaces

File: doc/manual/manual.tex [hunk 4]
```diff
@@ -2045,18 +2053,26 @@ \subsection{Installing and running \aspect{} on Mac OS X}
...
+\end{verbatim}
+
 \item Download  the  \aspect{} source from the git repository.
 
 {\footnotesize \begin{verbatim}
```
Comment (bangerth) [inline doc/manual/manual.tex]: while you're here, could you remove the `\footnotesize` thing? unless you recall why it's there in the first place...

File: doc/manual/manual.tex [hunk 5]
```diff
@@ -2045,18 +2053,26 @@ \subsection{Installing and running \aspect{} on Mac OS X}
...
 \end{verbatim}}
 
-
-Note: you {\bf MUST} \underline{build} and \underline{run} \aspect{} in the terminal window started by \dealii{}.
+If you wish to use Xcode as your IDE it is recommended to skip to step 7 below. Otherwise, proceed with steps 5 and 6 below.
 
 \item Build \aspect{}
 
+Note: you {\bf MUST} \underline{build} and \underline{run} \aspect{} in the terminal window started by \dealii{}.
+
```
Comment (bangerth) [inline doc/manual/manual.tex]: What does this comment refer to? That you can't run it in Xcode? Maybe that would be worth stating explicitly.

File: doc/manual/manual.tex [hunk 6]
```diff
@@ -2124,6 +2140,31 @@ \subsection{Installing and running \aspect{} on Mac OS X}
...
+cmake -G Xcode .
+\end{verbatim}
+
+Now open the resulting \texttt{aspect.xcodeproj} in Xcode.
+
+\item Open Product\textgreater Scheme\textgreater Manage Schemes, then select ``aspect'' and click ``Edit''. Go to Run\textgreater Arguments, and under "Arguments Passed On Launch" add the filepath to the desired parameter file. Click "Close" then select Product\textgreater Run to run \aspect{} with this parameter file.
+
+\item If you want to run in parallel in Xcode, the easiest way is to create a new scheme. Open Product\textgreater Scheme\textgreater Manage Schemes, select ``aspect'' then select the gear icon, then ``Duplicate scheme''. Rename the new scheme to ``aspect parallel''. Go to Run\textgreater Info, and in Executable choose ``Other...''. Navigate to 
+
+\begin{verbatim}
```
Comment (bangerth) [inline doc/manual/manual.tex]: no empty line here

File: doc/manual/manual.tex [hunk 7]
```diff
@@ -2124,6 +2140,31 @@ \subsection{Installing and running \aspect{} on Mac OS X}
...
+/Applications/deal.II.app/Contents/Resources/opt/openmpi-1.10.2/bin
+\end{verbatim}
+and select ``orterun'', then click ``Choose''.
+
+\item Now go to the Arguments tab, and add 3 arguments: first, add \texttt{-np4} (for 4 processors - you may use however many your machine has). Then add a path to your aspect executable - this should be within your aspect folder, or the subfolders Debug or Release. Finally, ensure you have a path to your parameter file as above. Ensure these are in the correct order, or reorder them if necessary. Click ``Close'' then select Product\textgreater Run to run \aspect{} in parallel with this parameter file.
```
Comment (bangerth) [inline doc/manual/manual.tex]: `-`  ->  `--`

Comment (bangerth) [inline doc/manual/manual.tex]: same in the second sentence

File: doc/manual/manual.tex (hunk 3 above)
Comment (spco) [inline doc/manual/manual.tex]: I'm not sure of the exact rules here - some seem to be indented by 2 spaces, some by 1, and some by none?

File: doc/manual/manual.tex (hunk 4 above)
Comment (spco) [inline doc/manual/manual.tex]: I think the `\footnotesize` is there to stop the lines going beyond the right margin - what is best to do here - keep it as is, or split the lines in some way?

File: doc/manual/manual.tex (hunk 3 above)
Comment (bangerth) [inline doc/manual/manual.tex]: In that case, do as you see fit, after looking at the manual.pdf.

File: doc/manual/manual.tex (hunk 4 above)
Comment (bangerth) [inline doc/manual/manual.tex]: Really, the lines here are so short anyway...

**Comment by spco** [comment]
So I've indented all verbatims by 2 spaces, but left `\footnotesize` where removing it would cause the line to go beyond the margin, and within the rest of that item to help readability. All other changes have been made, except this:

As far as compiler choices go, I'm not sure what to say - it certainly works with clang v7.3.0 - I could put in a remark to the effect that this is tested with 7.3.0? I don't know of a way to check what the .dmg was compiled with.

**Comment by bangerth** [comment]
Let's leave this issue for a later time. I'll open a bug report for it.

**Comment by spco** [comment]
Fixes #975 .

**Comment by bangerth** [comment]
Forgot to mention: can you please add your name to the list of the aspect manual in a separate pull request? (It's in doc/manual/manual.tex.)

**Pull request merged**